cmd/state-transition/state_transition.cpp

```diff
@@ -0,0 +1,318 @@
+#include "silkworm/core/protocol/rule_set.hpp"
+#include "silkworm/core/rlp/encode_vector.hpp"
+#include "silkworm/core/state/in_memory_state.hpp"
+#include "silkworm/sentry/common/ecc_key_pair.hpp"
+#include "third_party/ethash/include/ethash/keccak.hpp"
+
+namespace silkworm::cmd::state_transition {
+
+StateTransition::StateTransition(const std::string& file_path) noexcept {
+    std::ifstream input_file(file_path);
+    nlohmann::json baseJson;
+    input_file >> baseJson;
+    auto testObject = baseJson.begin();
+    test_name_ = testObject.key();
+    test_data_ = testObject.value();
+
+    terminate_on_error_ = false;
+    show_diagnostics_ = false;
+}
+
+StateTransition::StateTransition(const nlohmann::json& json, const bool terminate_on_error, const bool show_diagnostics) noexcept {
+    auto testObject = json.begin();
+    test_name_ = testObject.key();
+    std::cout << test_name_ << ":" << std::endl;
+
+    test_data_ = testObject.value();
+
+    terminate_on_error_ = terminate_on_error;
+    show_diagnostics_ = show_diagnostics;
+}
+
+std::string StateTransition::name() {
+    return test_name_;
+}
+
+std::string StateTransition::get_env(const std::string& key) {
+    return test_data_.at("env").at(key);
+}
+
+bool StateTransition::contains_env(const std::string& key) {
+    return test_data_.at("env").contains(key);
+}
+
+std::vector<ExpectedState> StateTransition::get_expected_states() {
+    std::vector<ExpectedState> expectedStates;
+
+    auto post = test_data_.at("post");
+
+    for (const auto& postState : post.items()) {
+        nlohmann::json data = postState.value();
+        const std::string& key = postState.key();
+        expectedStates.emplace_back(data, key);
+    }
+
+    return expectedStates;
+}
+
+evmc::address StateTransition::to_evmc_address(const std::string& address) {
+    evmc::address out;
+    if (!address.empty()) {
+        auto bytes = silkworm::from_hex(address);
+        out = silkworm::to_evmc_address(bytes.value_or(Bytes{}));
+    }
+
+    return out;
+}
+
+silkworm::Block StateTransition::get_block(protocol::IRuleSet& /*rule_set*/, InMemoryState& state, ChainConfig& chain_config) {
+    auto block = silkworm::Block();
+
+    block.header.beneficiary = to_evmc_address(get_env("currentCoinbase"));
+
+    block.header.gas_limit = std::stoull(get_env("currentGasLimit"), nullptr, /*base=*/16);
+    block.header.number = std::stoull(get_env("currentNumber"), nullptr, /*base=*/16);
+    block.header.timestamp = std::stoull(get_env("currentTimestamp"), nullptr, /*base=*/16);
+    block.header.parent_hash = to_bytes32(silkworm::from_hex(get_env("previousHash")).value_or(silkworm::Bytes{}));
+
+    if (contains_env("currentRandom")) {
+        block.header.prev_randao = to_bytes32(silkworm::from_hex(get_env("currentRandom")).value_or(silkworm::Bytes{}));
+    }
+
+    const evmc_revision rev{chain_config.revision(block.header.number, block.header.timestamp)};
+
+    //    if (rev <= EVMC_LONDON)
```

Comment:
It's better to remove this comment if it is a leftover or replace it with a `// TODO` if something needs to be done here